src/client/mod.rs

```diff
@@ -976,12 +976,12 @@ impl<T> VaultClient<T>
                                              endpoint: S1,
                                              wrap_ttl: Option<S2>)
                                              -> Result<Response> {
-        let mut req = self.client
-            .get(try!(self.host.join(endpoint.as_ref())))
-            .header(XVaultToken(self.token.to_string()))
-            .header(header::ContentType::json());
+        let h = self.host.join(endpoint.as_ref())?;
+        let mut req = self.client.get(h);
```

Comment:
This could look more clean like:

```rust
match wrap_ttl {
        Some(wrap_ttl) => {
            Ok(try!(handle_hyper_response(self.client
                .request(Method::Get, h)
                .header(XVaultToken(self.token.to_string()))
                .header(header::ContentType::json())
                .header(XVaultWrapTTL(wrap_ttl.into()))
                .send())))
        },
        None => {
            Ok(try!(handle_hyper_response(self.client
                .request(Method::Get, h)
                .header(XVaultToken(self.token.to_string()))
                .header(header::ContentType::json())
                .send())))
        }
    }
```

and that pattern could be continued in the other HTTP methods; It may be worth building out a bit of helper functionality to handle cleaning up the short lived borrows but this method does work ;-)